crates/network/src/any/mod.rs

```diff
@@ -66,11 +62,102 @@ impl Network for AnyNetwork {
 
     type TransactionRequest = WithOtherFields<TransactionRequest>;
 
-    type TransactionResponse = AnyRpcTransaction;
+    type TransactionResponse = WithOtherFields<Transaction<AnyTxEnvelope>>;
 
     type ReceiptResponse = AnyTransactionReceipt;
 
     type HeaderResponse = AnyRpcHeader;
 
     type BlockResponse = AnyRpcBlock;
 }
+
+/// A wrapper for [`AnyRpcBlock`] that allows for handling unknown block types.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct AnyRpcBlock(
+    WithOtherFields<Block<WithOtherFields<Transaction<AnyTxEnvelope>>, AnyRpcHeader>>,
```

Comment:
we also need from impls for these, you can use derive_more::From for all of these